apps/roam/src/components/LeftSidebarView.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useMemo, useState } from "react";
+import ReactDOM from "react-dom";
+import { Collapse, Icon } from "@blueprintjs/core";
+import getPageUidByPageTitle from "roamjs-components/queries/getPageUidByPageTitle";
+import openBlockInSidebar from "roamjs-components/writes/openBlockInSidebar";
+import extractRef from "roamjs-components/util/extractRef";
+import { getFormattedConfigTree } from "~/utils/discourseConfigRef";
+import type {
+  LeftSidebarConfig,
+  LeftSidebarPersonalSectionConfig,
+} from "~/utils/getLeftSidebarSettings";
+import { createBlock } from "roamjs-components/writes";
+import deleteBlock from "roamjs-components/writes/deleteBlock";
+import getTextByBlockUid from "roamjs-components/queries/getTextByBlockUid";
+
+const parseReference = (text: string) => {
+  const extracted = extractRef(text);
+  if (text.startsWith("((") && text.endsWith("))")) {
+    return { type: "block" as const, uid: extracted, display: text };
+  } else {
+    return { type: "page" as const, title: extracted, display: extracted };
+  }
+};
+
+const truncate = (s: string, max: number | undefined): string => {
+  if (!max || max <= 0) return s;
+  return s.length > max ? `${s.slice(0, max)}...` : s;
+};
+
+const openTarget = async (e: React.MouseEvent, sectionTitle: string) => {
+  e.preventDefault();
+  e.stopPropagation();
+  const target = parseReference(sectionTitle);
+
+  if (target.type === "block") {
+    if (e.shiftKey) {
+      await openBlockInSidebar(target.uid);
+      return;
+    }
+    await window.roamAlphaAPI.ui.mainWindow.openBlock({
+      block: { uid: target.uid },
+    });
+    return;
+  }
+
+  const uid = getPageUidByPageTitle(sectionTitle);
+  if (!uid) return;
+  if (e.shiftKey) {
+    await window.roamAlphaAPI.ui.rightSidebar.addWindow({
+      // @ts-expect-error - todo test
```

Comment:
.

Reply:
😅 I copied it from other parts of codebase, I was not sure what to do about it so I searched and copied it to atleast keep it consistent if someone searched for it in future